packages/block-library/src/navigation-link/edit.js

```diff
@@ -242,6 +242,18 @@ export default function NavigationLinkEdit( {
 	const itemLabelPlaceholder = __( 'Add link…' );
 	const ref = useRef();
 
+	const isCustomItemType =
```

Comment:
I think given that one of the built in types is `custom` it would be helpful to have a docblock here explaining more about what a "custom" link actually is.

Also perhaps consider renaming the variable to be even more explicit about [what this variable represents](https://developer.wordpress.org/reference/hooks/customize_nav_menu_available_item_types/) so as to distinguish it.

```suggestion
	const isCustomisedNavMenuItemType =
```

Finally you could consider extracting to a helper method with it's own function docblock...etc.

Reply:
With the changes in 42f7551040eaba0ef21c20e76f774db207ad22c0 this part of the code is no longer necessary.